### PR TITLE
Adding custom image to repo

### PR DESCRIPTION
**Pull request by norlandrhagen**
Adds CI infrastructure to build a custom image for this project.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -4,9 +4,9 @@ on:
   push:
     branches:
       - main
-  pull_request:
-    branches:
-      - main
```
Comment (andersy005) [inline .github/workflows/main.yml]: do we have to disable PR triggers? Are any limitations that necessitate this? i think it's crucial to be able to run the benchmarks on PRs.

File: binder/apt.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+
+libasound2
+libatk-bridge2.0-0
+libatk1.0-0
+libatspi2.0-0
+libcairo2
```
Comment (maxrjones) [inline binder/apt.txt:6]: what is the motivation for the extra packages relate to those listed by playwright (e.g., libcairo2, libpango-1.0-0)?

File: .github/workflows/main.yml (hunk 1 above)
Comment (norlandrhagen) [reply .github/workflows/main.yml]: Good catch! Was trying to cut down on the ci builds when playing with the image and forgot to comment that back in.

File: binder/apt.txt (hunk 2 above)
Comment (norlandrhagen) [reply binder/apt.txt:6]: I think I got a build error in playwright without those additional packages, but I'll double check.

Comment (norlandrhagen) [reply binder/apt.txt:6]:
```
║ Host system is missing dependencies to run browsers. ║
║ Please install them with the following command:      ║
║                                                      ║
║     playwright install-deps                          ║
║                                                      ║
║ Alternatively, use apt:                              ║
║     apt-get install libatk1.0-0\                     ║
║         libatk-bridge2.0-0\                          ║
║         libatspi2.0-0\                               ║
║         libcairo2                                    ║
║                                                      ║
║ <3 Playwright Team    
```

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+name: Build and push container image
+
+on:
+  pull_request:
+    paths:
+      - 'binder/**.yml'
+      - 'apt.txt'
```
Comment (maxrjones) [inline .github/workflows/build.yaml]:
```suggestion
      - 'binder/apt.txt'
```

**Review by maxrjones — APPROVED** [review]
Thanks @norlandrhagen!

**Review by andersy005 — APPROVED** [review]

**Pull request merged**